src/commands/data/import/bulk.ts

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as fs from 'node:fs';
+import { platform } from 'node:os';
+import { SfCommand, Flags } from '@salesforce/sf-plugins-core';
+import { MultiStageOutput } from '@oclif/multi-stage-output';
+import { Connection, Messages, Org } from '@salesforce/core';
+import { Duration } from '@salesforce/kit';
+import terminalLink from 'terminal-link';
+import { IngestJobV2, JobInfoV2 } from '@jsforce/jsforce-node/lib/api/bulk2.js';
+import { Schema } from '@jsforce/jsforce-node';
+import { BulkImportRequestCache } from '../../../bulkDataRequestCache.js';
+
+Messages.importMessagesDirectoryFromMetaUrl(import.meta.url);
+const messages = Messages.loadMessages('@salesforce/plugin-data', 'data.import.bulk');
+
+export type DataImportBulkResult = {
+  jobId: string;
+  processedRecords?: number;
+  successfulRecords?: number;
+  failedRecords?: number;
+};
+
+export default class DataImportBulk extends SfCommand<DataImportBulkResult> {
+  public static readonly summary = messages.getMessage('summary');
+  public static readonly description = messages.getMessage('description');
+  public static readonly examples = messages.getMessages('examples');
+
+  public static readonly flags = {
+    async: Flags.boolean({
+      summary: messages.getMessage('flags.async.summary'),
+      char: 'a',
+      exclusive: ['wait'],
+    }),
+    file: Flags.file({
+      summary: messages.getMessage('flags.file.summary'),
+      char: 'f',
+      required: true,
+      exists: true,
+    }),
+    sobject: Flags.string({
+      summary: messages.getMessage('flags.sobject.summary'),
+      char: 's',
+      required: true,
+    }),
+    'api-version': Flags.orgApiVersion(),
+    wait: Flags.duration({
+      summary: messages.getMessage('flags.wait.summary'),
+      char: 'w',
+      unit: 'minutes',
+      exclusive: ['async'],
+    }),
+    'target-org': Flags.requiredOrg(),
+    'line-ending': Flags.option({
+      summary: messages.getMessage('flags.line-ending.summary'),
+      dependsOn: ['file'],
+      options: ['CRLF', 'LF'] as const,
+    })(),
+  };
+
+  public async run(): Promise<DataImportBulkResult> {
+    const { flags } = await this.parse(DataImportBulk);
+
+    const conn = flags['target-org'].getConnection(flags['api-version']);
+
+    const timeout = flags.async ? Duration.minutes(0) : flags.wait ?? Duration.minutes(0);
+    const async = timeout.milliseconds === 0;
+
+    const baseUrl = flags['target-org'].getField<string>(Org.Fields.INSTANCE_URL).toString();
+
+    const ms = new MultiStageOutput<JobInfoV2>({
+      jsonEnabled: flags.json ?? false,
```

Comment:
you should use `this.jsonEnabled()` since there's some env var logic in there: https://github.com/salesforcecli/sf-plugins-core/blob/main/src/sfCommand.ts#L174